File: backend/app/domain/craving_trigger.py
from enum import Enum


class CravingTrigger(str, Enum):
    STRESS = "stress"
    BOREDOM = "boredom"
    AFTER_MEALS = "afterMeals"
    COFFEE = "coffee"
    HABIT = "habit"
    SOCIAL = "social"
    MORNING = "morning"
    OTHER = "other"  # fallback for old session rows

File: frontend/lib/features/session/domain/craving_trigger.dart
// `other` is not shown in the intake UI — it exists only as a fallback when
// parsing old session rows that used a trigger value no longer in the list.
enum CravingTrigger { stress, boredom, afterMeals, coffee, habit, social, morning, other }

extension CravingTriggerLabel on CravingTrigger {
  String get label {
    switch (this) {
      case CravingTrigger.stress:
        return 'Stress';
      case CravingTrigger.boredom:
        return 'Boredom';
      case CravingTrigger.afterMeals:
        return 'After meals';
      case CravingTrigger.coffee:
        return 'Coffee';
      case CravingTrigger.habit:
        return 'Habit';
      case CravingTrigger.social:
        return 'Social';
      case CravingTrigger.morning:
        return 'Morning';
      case CravingTrigger.other:
        return 'Other';
    }
  }
}
